testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -45,10 +46,28 @@
     private static final String TRINO_CATALOG = "iceberg";
     private static final String TEST_SCHEMA_NAME = "default";
 
-    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
-    public void testTrinoReadingSparkData()
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS}, dataProviderClass = TestingIcebergStorageFormat.class, dataProvider = "unsupportedStorageFormats")
```

Comment:
This is overkill. Iceberg by specification only has three formats. Hard-code this test to Avro and don't clutter `TestingIcebergStorageFormat` with supported/unsupported. This will make it easy to remove when we support Avro (which should be soon).

Reply:
This is specifically to self-test the `unsupportedStorageFormats`, otherwise we may easily forget to update the `@DataProviders` here when adding AVRO support.

Reply:
You could solve this by adding a single TODO comment to this test to remove this test and update TestingStoraheFormat when Avro is added. This test will fail at that point. No need to complicate all the code.

Reply:
>  No need to complicate all the code.


the only "complexity" is here, and i don't quite  understand why we're calling this a complexity at all.

i specifically named the other factory method "storageFormats", so that no change is needed when we add Avro support and can remove "unsupportedStorageFormats".